schemas/database/006_deletions.sql

```diff
@@ -0,0 +1,3 @@
+CREATE TABLE deletions (
+    leaf_index    BIGINT NOT NULL PRIMARY KEY
```

Comment:
You might consider adding a commitment column here. It'll make it easier to correlate the replacement identity

Reply:
Especially given the fact that `leaf_index` won't be unique any longer